executor/distsql.go

```diff
@@ -582,6 +594,19 @@ func (e *IndexLookUpExecutor) startIndexWorker(ctx context.Context, workCh chan<
 				worker.syncErr(err)
 				break
 			}
+			e.mu.Lock()
+			select {
+			case <-finishedCh:
+				finished = true
+			default:
+			}
+			if finished {
+				result.Close()
+				e.mu.Unlock()
+				break
+			}
+			e.pendingIndexResults = result
+			e.mu.Unlock()
```

Comment:
What's the difference check again here between checking in the next loop at line 578?